python/taichi/lang/transformer.py

```diff
@@ -485,16 +485,19 @@ def visit_FunctionDef(self, node):
           array_dim = self.arg_features[i][1]
           import numpy as np
           array_dt = to_taichi_type(array_dt)
-          if array_dt == ti.f32:
-            dt = self.parse_expr('ti.f32')
-          elif array_dt == ti.f64:
-            dt = self.parse_expr('ti.f64')
-          elif array_dt == ti.i32:
-            dt = self.parse_expr('ti.i32')
-          elif array_dt == ti.i64:
-            dt = self.parse_expr('ti.i64')
-          else:
-            assert False
+          dt_table = {
+              ti.f32: 'ti.f32',
+              ti.f64: 'ti.f64',
+              ti.i8: 'ti.i8',
+              ti.i16: 'ti.i16',
+              ti.i32: 'ti.i32',
+              ti.i64: 'ti.i64',
+              ti.u8: 'ti.u8',
+              ti.u16: 'ti.u16',
+              ti.u32: 'ti.u32',
+              ti.u64: 'ti.u64',
+          }
+          dt = self.parse_expr(dt_table[array_dt])
```

Comment:
You may want more like this for adding new data types!

Reply:
Cool!

Reply:
We can also make use of `ti.core.data_type_short_name`.